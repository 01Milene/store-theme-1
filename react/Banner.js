import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { Link } from 'render'

class Banner extends Component {
  static propTypes = {
    image: PropTypes.string,
    page: PropTypes.string,
    targetParams: PropTypes.object,
  }

  static schema = {
    title: 'Banner',
    description: 'A simple banner component',
    type: 'object',
    properties: {
      image: {
        type: 'string',
        title: 'Image',
      },
      page: {
        type: 'string',
        title: 'Link',
      },
      targetParams: {
        type: 'object',
        title: 'Target params',
        properties: {
          params: {
            type: 'string',
            title: 'Params',
          },
        },
      },
    },
  }

  render() {
    const { image, page, targetParams } = this.props
    return (
      <div className="flex items-center">
        <Link className="mb0" page={page} params={targetParams}>
          <img src={image} />
        </Link>
      </div>
    )
  }
}

export default Banner
