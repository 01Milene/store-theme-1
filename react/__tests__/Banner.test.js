import React from 'react'
import { shallow } from 'enzyme'
import Banner from '../Banner'

describe('Banner test', () => {
  let wrapper

  beforeEach(() => {
    const image = ''
    wrapper = shallow(<Banner image={image} />)
  })

  it('should render correctly', () => {
    expect(wrapper.getElement()).toMatchSnapshot()
  })
})
