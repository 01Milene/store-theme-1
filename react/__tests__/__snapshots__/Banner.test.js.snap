// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner test should render correctly 1`] = `
<div>
  <a>
    <img
      src=""
    />
  </a>
</div>
`;
